plugins/org.python.pydev.debug/src/org/python/pydev/debug/ui/PathMappingsPrefsPage.java

```diff
@@ -0,0 +1,168 @@
+
+    /**
+     * Initializer sets the preference store
+     */
+    public PathMappingsPrefsPage() {
+        super("Path Mapping", GRID);
+        setPreferenceStore(PydevPlugin.getDefault().getPreferenceStore());
+    }
+
+    @Override
+    public void init(IWorkbench workbench) {
+    }
+
+    /**
+     * Creates the editors
+     */
+    @Override
+    protected void createFieldEditors() {
+        Composite p = getFieldEditorParent();
+        jsonFieldEditor = new JsonFieldEditor(PyDevEditorPreferences.PATHS_FROM_ECLIPSE_TO_PYTHON,
+                "Path Mappings JSON input", p);
+        jsonFieldEditor.setAdditionalJsonValidation((json) -> checkPathMappingsFormat(json));
+        addField(jsonFieldEditor);
+
+        addTemplateButton = new Button(p, SWT.PUSH);
+        addTemplateButton.addSelectionListener(new SelectionListener() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                try {
+                    addTemplateButtonClick();
+                } catch (InterruptedException e1) {
+                    Log.log(e1);
+                }
+            }
+
+            @Override
+            public void widgetDefaultSelected(SelectionEvent e) {
+            }
+        });
+        addTemplateButton.setText("Add path mapping template entry.");
+    }
+
+    private void addTemplateButtonClick() throws InterruptedException {
+        if (jsonFieldEditor != null) {
+            StyledText textField = jsonFieldEditor.getTextControl(getFieldEditorParent());
+            if (textField != null && !textField.isDisposed()) {
+                FastStringBuffer contentBuffer = new FastStringBuffer();
+                contentBuffer.append(textField.getText()).trim();
+                if (contentBuffer.isEmpty()) {
+                    textField.setText(pathMappingsListTemplate);
+                } else if (jsonFieldEditor.isValidToApply()) {
+                    contentBuffer.deleteLast();
+                    contentBuffer.trim().append(pathMappingsObjectTemplate);
+                    textField.setText(contentBuffer.toString());
+                }
+            }
+        }
+    }
+
+    @Override
+    protected void updateApplyButton() {
+        super.updateApplyButton();
+        if (isValid()) {
+            addTemplateButton.setEnabled(true);
+        } else {
+            addTemplateButton.setEnabled(false);
+        }
+    }
+
+    private Optional<String> checkPathMappingsFormat(JsonValue json) {
+        if (!json.isArray()) {
+            return Optional.of("Path Mappings JSON must be an array.");
+        }
+        JsonArray array = json.asArray();
+        if (array.size() == 0) {
+            return Optional.of("Empty array.");
+        }
+
+        int i = 0;
+        for (JsonValue value : array) {
+            if (!value.isObject()) {
+                return Optional.of("Only objects are accepted.");
+            }
+
+            i++;
+            String objectId = "Object " + i;
```

Comment:
I think the object id could be:

`String objectId = "Path mapping entry: " + i`;